federated_content_connector/course_metadata_importer.py

```diff
@@ -0,0 +1,242 @@
+            cls.store_courses_details(processed_courses_details)
+
+            logger.info(f'[COURSE_METADATA_IMPORTER] Import completed. Courses: {courserun_keys}')
+
+        logger.info('[COURSE_METADATA_IMPORTER] Course metadata import completed for all courses.')
+
+    @classmethod
+    def courserun_locators_to_import(cls):
+        """
+        Construct list of active course locators for which we want to import data.
+
+        We will exclude the courseruns which are already imported.
+        """
+        course_overviews = CourseOverview.get_all_courses()
+        course_details_ids = list(CourseDetails.objects.all().values_list('id', flat=True))
+
+        logger.info(
+            f'[COURSE_METADATA_IMPORTER] Already imported courseruns will be excluded. Keys: {course_details_ids}'
+        )
+
+        now = datetime.datetime.now(pytz.UTC)
+        return list(course_overviews.filter(
+            Q(end__gt=now) &
+            (
+                Q(enrollment_end__gt=now) |
+                Q(enrollment_end__isnull=True)
+            )
+        ).exclude(id__in=course_details_ids).values_list(
+            'id',
+            flat=True
+        ))
+
+    @classmethod
+    def fetch_courses_details(cls, client, courserun_locators, api_base_url):
+        """
+        Fetch the course data from discovery using `/api/v1/courses` endpoint.
+        """
+        course_keys = [cls.construct_course_key(courserun_locator) for courserun_locator in courserun_locators]
+        encoded_course_keys = ','.join(map(quote_plus, course_keys))
+
+        logger.info(f'[COURSE_METADATA_IMPORTER] Fetching details from discovery. Courses {course_keys}.')
+        api_url = f"{api_base_url}/courses/?keys={encoded_course_keys}"
+        response = client.get(api_url)
+        response.raise_for_status()
+        courses_details = response.json()
+        results = courses_details.get('results', [])
+
+        # Find and log the course keys not found in course-discovery
+        course_keys_in_response = [result.get('key') for result in results]
+        courses_not_found = list(set(course_keys) - set(course_keys_in_response))
+        if courses_not_found:
+            logger.info(f'[COURSE_METADATA_IMPORTER] Courses not found in discovery. Courses: {courses_not_found}')
+
+        return results
+
+    @classmethod
+    def process_courses_details(cls, courserun_locators, courses_details):
+        """
+        Parse and extract the minimal data that we need.
+        """
+        courses = {}
+        for courserun_locator in courserun_locators:
+            course_key = cls.construct_course_key(courserun_locator)
+            courserun_key = str(courserun_locator)
+            course_metadata = cls.find_attr(courses_details, 'key', course_key)
+            if not course_metadata:
+                continue
+
+            course_type = course_metadata.get('course_type') or ''
+            product_source = course_metadata.get('product_source') or ''
```

Comment:
[curious] Is there any risk in having the fallback value be `None` instead of an empty string?

Reply:
product_source is a CharField on model and it has null=False as per [Django's recommendation](https://docs.djangoproject.com/en/3.2/ref/models/fields/#null) That is why we are setting an empty string in case of no value.